src/Page/Account.elm

```diff
@@ -1,50 +1,237 @@
 module Page.Account exposing (Model, Msg(..), init, update, view)
 
-import Data.Account exposing (Account)
-import Data.Transaction exposing (LedgerEntry, decodeLedgerEntry)
+import Data.Account exposing (Account, AccountId)
+import Data.Transaction as Transaction exposing (LedgerEntry, TransactionId, decodeLedgerEntry)
 import DateFormat
 import Dict exposing (Dict)
-import Element exposing (..)
+import Editable exposing (Editable(..))
+import Element exposing (Attribute, Element, column, el, fill, fillPortion, html, row, text, width)
+import Element.Events exposing (onClick)
+import Element.Input as Input exposing (focusedOnLoad)
+import FormValue exposing (FormValue(..))
 import Http
 import Json.Decode as Decode
+import List.Extra as LE
+import Loadable exposing (Loadable(..))
 import Time
+import Tree
+import UI exposing (accountSelect, columnRow, formValueEdit, onEnter)
 import Utils
 
 
 type Msg
     = RequestLedgerEntries Account
-    | ReceiveLedgerEntries (Result Http.Error (List LedgerEntry))
+    | ReceiveLedgerEntries (Result Http.Error (List LedgerEntryForm))
+    | EditLedgerEntry TransactionId
+    | ChangeLedgerEntryDescription TransactionId String
+    | ChangeLedgerEntryOtherSplitAmount TransactionId String
+    | ChangeLedgerEntryOwnSplitAmount TransactionId String
+    | ChangeLedgerEntryAccount TransactionId AccountId
+    | SaveLedgerEntry (Editable LedgerEntryForm)
+    | SavedLedgerEntry (Result Http.Error LedgerEntryForm)
+
+
+
+-- MODEL
 
 
 type alias Model =
     { account : Account
-    , ledgerEntries : List LedgerEntry
-    , lastError : String
+    , ledgerEntries : Loadable (List (Editable LedgerEntryForm)) ()
```

Comment:
I think so. Ledger entries are loaded from the API. `Loadable` captures the pattern where something is loading and then loaded or failed along with showing progress. Why wouldn't we want to reuse the functor here instead of handling it using a new abstraction?

Reply:
Ok, makes sense.